src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/StubbedProductDefinitionRepository.kt

```diff
@@ -220,4 +222,11 @@ class StubbedProductDefinitionRepository : ProductDefinitionRepository {
       ),
     )
   }
+
+  override fun getDataSet(reportId: String, dataSetId: String): DataSet {
+    return getProductDefinitions().filter { it.id == reportId }
+      .flatMap { pd -> pd.dataSet.filter { it.id == dataSetId } }
+      .ifEmpty { throw ValidationException("Invalid dataSetId provided: $dataSetId") }
+      .takeIf { it.size == 1 }?. first() ?: throw DuplicateDataSetIdException(dataSetId)
```

Comment:
I think once the definition is put into the appropriate data store, it will be the job of whatever adds the definition to validate it. This feels like a redundant validation in that context.

Also, the definition and variant IDs are not validated, which makes this a bit inconsistent.

Reply:
I have now removed the validation which throws the DuplicateDataSetIdException.